rfcs/inline_symbols.md

```diff
@@ -0,0 +1,624 @@
+ |  +--------------------------- VarInt -6: 6 bytes of inline text
+ +------------------------------ Singleton annotation	
+```
+
+or using a symbol ID:
+
+```
+             E  r  n  e  s  t     H  e  m  i  n  g  w  a  y
+E1 CA 8E 90 45 72 6E 65 73 74 20 48 65 6D 69 6E 67 77 61 79 
+ |  |  |  +---- VarUInt Length: 16 bytes
+ |  |  +------- String w/VarUInt Length
+ |  +---- VarInt 10: symbol ID $10
+ +------- Singleton annotation
+```
+
+To skip a singleton annotation, the reader must read the `VarInt`. If it is negative, the reader
+must skip that number of bytes to move beyond the UTF8 bytes that follow it. The reader will then
+be positioned over the annotated value, which it can read or skip as needed.
+
+##### `0xE2`: Multiple inline symbol annotations
+
+```
+            7       4 3       0
+            +---------+---------+
+Value with  |   14    |    2    |
+multiple    +---------+---------+=======================================+
+annotations :     length [VarUInt]                                      :
+            +-----------------------------------------------------------+
+            :     annotation [VarInt + optional UTF8 representation]    : ...
+            +-----------------------------------------------------------+
+            :     value                                                 :
+            +-----------------------------------------------------------+
+```
+
+Unlike the singleton annotation encoding, the encoding for a value with multiple annotations
+includes a `length` field that indicates the number of bytes that will be used to represent
+the sequence of annotations that follow. It does not include the length of the value itself,
+which provides a length encoding of its own.
+
+Annotations in the sequence do not need to be encoded homogenously; writers can write some
+annotations as symbol IDs and others as inline text. (See the section [*Inline symbol
+structs*](#0xf4-inline-symbol-structs) for more detail.)
+
+To skip an annotation sequence, the reader must read the `VarUInt` `length` and skip that number of
+bytes. The reader will then be positioned over the annotated value, which it can read or skip as
+needed.
+
+### Inline symbols' relationship to Shared Symbol Tables
+
+Ion 1.0 provides [Shared Symbol
+Tables](http://amzn.github.io/ion-docs/docs/symbols.html#shared-symbol-tables) (SSTs)
+as a mechanism for reducing the overhead of defining a set of symbols in a stream.
+
+Here we revisit our example weather station data from the [Short-lived
+streams](#short-lived-streams) section, which used a Local Symbol Table to define all of the symbols
+it needed at the beginning of the stream:
+
+```js
+$ion_1_0
+$ion_symbol_table::{ // Add all of our symbols, field names, and annotations to the symbol table
+  sensorId,   // $10
+  type,       // $11
+  sensorData, // $12
+  reading,    // $13
+  temperature,// $14
+  celsius,    // $15
+  time,       // $16
```

Comment:
Shouldn't this be `symbols: [ ... ]`?

Actually, this makes me wonder if we need or have text to emphasize that inline symbols are not allowed as symbol values in the table.  The text in the [symbols doc][1] says:

>The symbols field should be a list of strings. If the field is missing or has any other type, it is treated as if it were an empty list.

[1]: https://amzn.github.io/ion-docs/docs/symbols.html#local-symbol-tables

Reply:
> Shouldn't this be symbols: [ ... ]?

Good catch! This was a text editing flub on my part, not a bold new proposal 😛. I think the existing text in the symbols doc is sufficient.